Userland/Libraries/LibChess/UCIEndpoint.cpp

```diff
@@ -53,15 +53,30 @@ void Endpoint::event(Core::Event& event)
         return handle_bestmove(static_cast<BestMoveCommand const&>(event));
     case Command::Type::Info:
         return handle_info(static_cast<InfoCommand const&>(event));
+    case Command::Type::Quit:
+        return handle_quit();
     default:
+        Object::event(event);
         break;
     }
 }
 
+void Endpoint::custom_event(Core::CustomEvent& custom_event)
+{
+    if (custom_event.custom_type() == EndpointEventType::UnexpectedEof)
+        handle_unexpected_eof();
+}
+
 void Endpoint::set_in_notifier()
 {
     m_in_notifier = Core::Notifier::construct(m_in->fd(), Core::Notifier::Read);
     m_in_notifier->on_ready_to_read = [this] {
+        if (m_in->eof()) {
+            Core::EventLoop::current().post_event(*this, make<Core::CustomEvent>(EndpointEventType::UnexpectedEof));
+            m_in_notifier->close();
```

Comment:
This shouldn't be necessary, `m_in` will automatically close the fd.

Reply:
Not calling `close()` explicitly will crash the GUI if the `ChessEngine` process is unexpectedly terminated.

Reply:
Looks to me like we are doing something wrong somewhere else and closing here just paper over the issue.

Reply:
I've changed this line from `m_in_notifier->close()` to `m_in_notifier->set_enabled(false)`, as I think that better shows our intent here. Not doing this will cause the notifier to spin on its `on_ready_to_read` callback. Several other places disable the notifier like this on EOF, so I think this is fine.

Reply:
Fine, good enough for me!